src/MPoly.jl

```diff
@@ -486,6 +486,31 @@ function is_monomial(x::MPolyRingElem{T}) where T <: RingElement
    return length(x) == 1 && isone(first(coefficients(x)))
 end
 
+@doc raw"""
+    is_homogeneous(x::MPolyRingElem{T}) where T <: RingElement
+
+Return `true` if the given polynomial is homogeneous with respect to the standard grading and `false` otherwise.
+"""
+function is_homogeneous(x::MPolyRingElem{T}) where {T <: RingElement}
+   last_deg = 0
+   is_first = true
+
+   for e in exponent_vectors(x)
+      d = sum(e)
+      if !is_first
+         if d != last_deg
+            return false
+         else
+            last_deg = d
+         end
+      else
+         is_first = false
+         last_deg = d
+      end
+   end
+   return true
+end
+
```

Comment:
Why is this in here? Following the discussion in #1649 I think "we" don't want this?

Reply:
I widened the signature just to be able to run more of the testsuite (since `is_homogeneous` for UnivPoly calls it on the underlying polynomial, and that errors for `ZZMPolyElem` from Nemo). But I was not aware of #1649, and so will revert all changes around `is_homogeneous`. Thanks for pointing it out.